### ISSUE TITLE
MERGE semantic should ignore duplicate records (bug from a COPY FROM CSV statement)

### ISSUE THREAD

**Issue by hs41-18** [kuzudb/kuzu#2600]
hi
let's say I have a CSV file containing some operational data with these headers:

Region	Country	
NA		US 
APAC	CHINA	
EMEA 	UK	

, im trying to work out the query to copy nodes and relations from a single CSV.  An equivalent neo4j query runs like this:

LOAD CSV WITH HEADERS FROM _file_ AS row
MERGE (r:Region {region: row.Region})
MERGE (c:Country {country: row.Country})
MERGE (c)-[:IN_REGION]->(r)

as I already have region, country, and in_region rel tables available, I'd expect the above query to run in Kuzu as well.

**Comment by andyfengHKU** [comment]
Hi @hs41-18 

We use a slightly different syntax when loading from a CSV file, that is, Kùzu directly parse a row into multiple columns. Below is a sequence of statements for loading. I assume the delimiter is `,` in your CSV file.
```
CREATE NODE TABLE Region (region STRING, PRIMARY KEY(region));
CREATE NODE TABLE Country (country STRING, PRIMARY KEY(country));
CREATE REL TABLE IN_REGION(FROM Country TO Region);

LOAD FROM <file> (header=true)
MERGE (r:Region {region: Region})
MERGE (c:Country {country: Country})
MERGE (c)-[:IN_REGION]->(r) ;
```

If you have large CSV files, I would suggest first format it into node files and relationship files whose bulk loading performance should be order of magnitude faster than `MERGE`. An example is shown below
```
COPY (LOAD FROM <file> (header=true) RETURN DISTINCT Region) TO "region.csv";
COPY (LOAD FROM <file> (header=true) RETURN DISTINCT Country) TO "country.csv";
COPY (LOAD FROM <file> (header=true) RETURN Country, Region) TO "in_region.csv";

COPY Region FROM "region.csv";
Copy Country FROM "country.csv";
Copy IN_REGION FROM "in_region.csv";
```

**Comment by hs41-18** [comment]
thanks for the reply @andyfengHKU. I want to achieve the first approach because my source file will contain ~50 columns, where I need to declare at least around 24 combinations of the columns, which is not easy challenge due to the size of the file. 
LOAD FROM <file> (header=true)
MERGE (r:Region {region: Region})
MERGE (c:Country {country: Country})
MERGE (c)-[:IN_REGION]->(r) ;

This approach seems to be working, apart from the natural duplication I got because of repetitive regions with different countries. What is the best way of using Distinct with this syntax?

**Comment by andyfengHKU** [comment]
Hi @hs41-18, seems we have a bug here because `MERGE` semantic should ignore duplicate records. I'll open an issue for this. 

Meanwhile the workaround that comes into my mind is
```
// For node tables
LOAD FROM (header=true)
WITH DISTINCT Region
CREATE (r:Region {region: Region})
...

// For rel table
LOAD FROM (header=true)
MATCH (r:Region {region: Region})
MATCH (c:Country {country: Country})
CREATE (c)-[:IN_REGION]->(r) ;
```

**Comment by hs41-18** [comment]
thank you @andyfengHKU  this is helpful, and happy to uncover a bug to fix. appreciate if you ping this thread once it is done

**Comment by andyfengHKU** [comment]
Forgot to mention but this should be fixed in #3216
